tests/topotests/multicast_pim6_static_rp_topo1/test_multicast_pim6_static_rp1.py

```diff
@@ -0,0 +1,1317 @@
+    logger.info("=" * 40)
+
+
+#####################################################
+#
+#   Local API
+#
+#####################################################
+
+
+def verify_state_incremented(state_before, state_after):
+    """
+    API to compare interface traffic state incrementing
+
+    Parameters
+    ----------
+    * `state_before` : State dictionary for any particular instance
+    * `state_after` : State dictionary for any particular instance
+    """
+
+    for router, state_data in state_before.items():
+        for state, value in state_data.items():
+            if state_before[router][state] >= state_after[router][state]:
+                errormsg = (
+                    "[DUT: %s]: state %s value has not"
+                    " incremented, Initial value: %s, "
+                    "Current value: %s [FAILED!!]"
+                    % (
+                        router,
+                        state,
+                        state_before[router][state],
+                        state_after[router][state],
+                    )
+                )
+                return errormsg
+
+            logger.info(
+                "[DUT: %s]: State %s value is "
+                "incremented, Initial value: %s, Current value: %s"
+                " [PASSED!!]",
+                router,
+                state,
+                state_before[router][state],
+                state_after[router][state],
+            )
+
+    return True
+
+
+#####################################################
+#
+#   Testcases
+#
+#####################################################
+
+
+def test_pim6_add_delete_static_RP_p0(request):
+    """
+    Verify upstream interfaces(IIF) and join state are updated
+        properly after adding and deleting the static RP
+    Verify IIF and OIL in "show ipv6 pim state" updated properly when
+        RP becomes unreachable
+    Verify RP becomes reachable after MLD join received, PIM join
+               towards RP is sent immediately
+
+    TOPOlogy used:
+         r0------r1-----r2
+       iperf    DUT     RP
+    """
+
+    tgen = get_topogen()
+    tc_name = request.node.name
+    write_test_header(tc_name)
+
+    # Don"t run this test if we have any failure.
+    if tgen.routers_have_failure():
+        check_router_status(tgen)
+
+    step("Creating configuration from JSON")
+    kill_socat(tgen)
+    clear_pim6_mroute(tgen)
+    clear_pim6_interface_traffic(tgen, TOPO)
+    reset_config_on_routers(tgen)
+
+    step("Shut link b/w R1 and R3 and R1 and R4 as per tescase topology")
+    intf_r1_r3 = TOPO["routers"]["r1"]["links"]["r3"]["interface"]
+    intf_r1_r4 = TOPO["routers"]["r1"]["links"]["r4"]["interface"]
+    for intf in [intf_r1_r3, intf_r1_r4]:
+        shutdown_bringup_interface(tgen, "r1", intf, ifaceaction=False)
+
+    step("Enable PIM6 between r1 and r2")
+    step("Enable MLD on r1 interface and send MLD " "join (FF08::1) to r1")
```

Comment:
Can we reuse already defined constants, like GROUP_ADDRESS_1?

Reply:
Done